packages/engine/utils/token.ts

```diff
@@ -0,0 +1,42 @@
+import { modules } from '@walless/ioc';
+import type { TokenDocument } from '@walless/store';
+
+const getTokenById = async (id: string): Promise<TokenDocument | undefined> => {
+	return await modules.storage.safeGet(id);
+};
+
+const setTokens = (tokens: TokenDocument[]) => {
+	const tokenPromises: Promise<unknown>[] = [];
+
+	for (const token of tokens) {
+		tokenPromises.push(modules.storage.upsert(token._id, async () => token));
+	}
+
+	Promise.all(tokenPromises);
+};
```

Comment:
I prefer name it `create` or `insert`, use `setTokens` makes me think it's a set state action

Reply:
could use `addTokensToStorage`, I guess?

Reply:
if using this name, we could rename all of functions make change storage make adding suffix `ToStorage1